cmd/controller-manager/controller-manager.go

```diff
@@ -22,13 +22,20 @@ import (
 	"k8s.io/component-base/cli"
 	"k8s.io/component-base/logs"
 	_ "k8s.io/component-base/logs/json/register" // To enable JSON log format support
+	"k8s.io/klog/v2"
 	controllerruntime "sigs.k8s.io/controller-runtime"
 
 	"github.com/karmada-io/karmada/cmd/controller-manager/app"
 )
 
 func main() {
 	ctx := controllerruntime.SetupSignalHandler()
+	// Starting from version 0.15.0, controller-runtime expects its consumers to set a logger through log.SetLogger.
+	// If SetLogger is not called within the first 30 seconds of a binaries lifetime, it will get
+	// set to a NullLogSink and report an error. Here's to silence the "log.SetLogger(...) was never called; logs will not be displayed" error
+	// by setting a logger through log.SetLogger.
+	// More info refer to: https://github.com/karmada-io/karmada/pull/4885.
+	controllerruntime.SetLogger(klog.Background())
```

Comment:
Joe, do you have any preference about the location where set the logger?
I'm asking because of https://github.com/karmada-io/karmada/pull/6430/files#r2153528865.

It seems to set the logger after ValidateAndApply is a good idea.

Reply:
Yeah.